src/python-common/ceph/deployment/drive_selection/selector.py

```diff
@@ -134,11 +134,19 @@ def assign_devices(self, device_filter):
             if disk in devices:
                 continue
 
-            if not all(m.compare(disk) for m in FilterGenerator(device_filter)):
-                logger.debug(
-                    "Ignoring disk {}. Filter did not match".format(
-                        disk.path))
-                continue
+            if self.spec.filter_logic == 'AND':
```

Comment:
The spec should already be validated before entering this method, but it would still be nice to check allowed `filter_logic` values here. Maybe an assertion?

Reply:
> The spec should already be validated before entering this method, but it would still be nice to check allowed `filter_logic` values here. Maybe an assertion?

Is the dashboard directly accessing `DriveSelection`? 

For any other orchestrator we have the access handled over `DriveGroupSpec` which does checks for valid arguments.

See:

https://github.com/ceph/ceph/pull/35916/files#diff-47814198f3158b86d01a0026e12bb3d6R304

Reply:
> Is the dashboard directly accessing `DriveSelection`?

No. It only calls `apply_drivegroups`. So the input spec defnintely will be checked.

Reply:
cool :+1: